### DISCUSSION TITLE
PHP-Parser limitation

### DISCUSSION THREAD

**Discussion by villfa** [rectorphp/rector#6874] [Q&A]
Hi,

I was trying to write a new rule for people migrating to PHP 8 to cover this RFC: https://wiki.php.net/rfc/namespaced_names_as_token
(see also https://3v4l.org/Yonll)

The goal was to transform the code this way:
```diff
- namespace Foo \ Bar;
+ namespace Foo\Bar;

- new \ stdClass();
+ new \stdClass();
```

Unfortunately PHP-Parser doesn't support this syntax anymore:  https://github.com/nikic/PHP-Parser/issues/700#issuecomment-683299766

Since I don't know what to do with it, I wanted to keep a trace about that at least.

**Comment by TomasVotruba** [comment]
Hi, this is job for a coding standard tool.

You can see the difference between tokens and AST in examples here:

* https://tomasvotruba.com/blog/2018/10/25/why-ast-fixes-your-coding-standard-better-than-tokens/
* https://tomasvotruba.com/blog/2018/10/22/brief-history-of-tools-watching-and-changing-your-php-code/
